presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -380,7 +380,8 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
         Path targetPath = locationService.targetPathRoot(locationHandle);
         createDirectory(session.getUser(), hdfsEnvironment, targetPath);
 
-        createTable(schemaName, tableName, tableMetadata.getOwner(), columnHandles, hiveStorageFormat, partitionedBy, bucketProperty, additionalTableParameters, targetPath);
+        Table table = createTable(schemaName, tableName, tableMetadata.getOwner(), columnHandles, hiveStorageFormat, partitionedBy, bucketProperty, additionalTableParameters, targetPath);
+        metastore.createTable(table);
     }
 
     private Table createTable(
```

Comment:
Rename this method `buildTableObject` or similar since it no longer creates the table